### ISSUE TITLE
[BUG] Multiple Monitor Docking Issue

### ISSUE THREAD

**Issue by castanley** [micheleg/dash-to-dock#319]
This error is a bit weird to explain, however it only occurs when Dash-To-Dock is enabled. When moving the dash/dock to a secondary monitor and moving windows to that monitor, the application window can only go so far and appears to be bound by some sort of forcefield ;)

When resizing the application itself, you are able to move the window more - as if decreasing the forcefield range. Anyway, there appears to be a direct correlation between the boundaries in which you can move and the size of the window itself. 

Also, when an application window is in full-screen - the dash/dock appears to overlap the application.

Here is a video of the bug in action as this is much easier to see what is going on rather than try to explain it: https://youtu.be/yym6afUCrms

$ uname -a
Linux localhost.localdomain 3.10.0-327.18.2.el7.x86_64 #1 SMP Thu May 12 11:03:55 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

$ cat /etc/centos-release
CentOS Linux release 7.2.1511 (Core)

**Comment by micheleg** [comment]
Mmmm, very weird. 

What extension and gnome version are you using?
Please confirm if:

* the problem occurs with other extensions disabled
* does the problem occurs only on with the secondary monitor on the right?
* can you guess what this "force field" could correspond to (like the dash height, half the window size, ....)?
* did the problem appear after a specific update?

**Comment by castanley** [comment]
So I have uninstalled all extensions except for Dash-to-Dock and it still occurs. 

$ gnome-shell --version
GNOME Shell 3.14.4

It appears that the "forcefield" bug only occurs when I use a monitor in portrait mode instead of having all the monitors in landscape mode AND when I have the dock bound to the monitor that is in portrait mode (right monitor). When changing the dock to be on my middle monitor (as depicted in the video) which is in landscape mode, this issue appears to be gone. However, the other issue mentioned is still present. When I have two monitors in landscape and the far right monitor in portrait (order doesn't matter) then the dock appears to be "floating" meaning that if I full screen any window the dock overlaps it causing part of the window to be obstructed. If I take my monitor out of portrait mode and use all of the monitors in landscape mode BOTH bugs disappear.

Just noticed it, I haven't used Dash-to-Dock in the past.

**Comment by micheleg** [comment]
Yes, I can reproduce the bug when the additional monitor is larger than the main one.  It's not clear to me the origin of the problem, there have always been issue with multiple monitors configurations. This is a quite strange one. I'll try to solve it as soon as possible.

Possibly related:
* https://git.gnome.org/browse/mutter/commit/?id=a7350475e846116d98cc110c2afa10151007b183
* https://git.gnome.org/browse/gnome-shell/commit/?h=gnome-3-14&id=585930123d2f411f7e6381d791bde3863c5f5af3
